grails-app/services/au/org/emii/netcdfsubset/SubsetService.groovy

```diff
@@ -12,17 +12,30 @@ class SubsetService {
         def generator = _getGenerator()
 
         try {
+            _writeTemplateToLayerConfigDir(typeName)
             generator.write(typeName, cqlFilter, dataSource.connection, response)
         }
         catch (Exception e) {
-            log.error "An error has occurred when writing netcdf file", e
+            log.error "An error has occurred when generating netcdf file", e
         }
     }
 
     def _getGenerator = { ->
          new NcdfGenerator(
-                grailsApplication.config.netcdf_filters.layer_config_dir,
-                System.getProperty('java.io.tmpdir')
+             grailsApplication.config.ncdfgenerator.layerConfigDir,
+             System.getProperty('java.io.tmpdir')
         )
     }
+
+    def _writeTemplateToLayerConfigDir(typeName) {
```

Comment:
I don't really feel comfortable about this - I'd be happier if perhaps the API to the generator was changed in either of the following ways:

1. the `layerConfigDir` parameter to the constructor was completely removed - let the upstream manage things itself (for now - we can make it configurable later - if and when it's needed)
2.the call to `write` took a java stream referencing a specific template, rather than a directory

Seems like the way it is now is a bit over complicated - the service it telling the upstream library things it already knows.